controller/users.go

```diff
@@ -99,21 +99,22 @@ func (c *UsersController) copyExistingKeycloakUserProfileInfo(ctx context.Contex
 		return nil, err
 	}
 
-	if existingProfile.FirstName != nil && keycloakUserProfile.FirstName != nil {
+	// If the *new* FirstName has already been set, we won't be updating it with the *existing* value
+	if existingProfile.FirstName != nil && keycloakUserProfile.FirstName == nil {
 		keycloakUserProfile.FirstName = existingProfile.FirstName
 	}
-	if existingProfile.LastName != nil && keycloakUserProfile.LastName != nil {
+	if existingProfile.LastName != nil && keycloakUserProfile.LastName == nil {
 		keycloakUserProfile.LastName = existingProfile.LastName
 	}
-	if existingProfile.Email != nil && keycloakUserProfile.Email != nil {
+	if existingProfile.Email != nil && keycloakUserProfile.Email == nil {
 		keycloakUserProfile.Email = existingProfile.Email
 	}
-	if existingProfile.Attributes != nil && keycloakUserProfile.Attributes != nil {
+	if existingProfile.Attributes != nil && keycloakUserProfile.Attributes == nil {
 		// If there are existing attributes, we overwite only those
 		// handled by the Users service in platform.
 		keycloakUserProfile.Attributes = existingProfile.Attributes
```

Comment:
My undertanding of this block is that you're going to set the Attributes in the KC user profile with the exiting profile attributes if the former are not set yet. I don't think this corresponds to the comment on lines 113/114